### PR TITLE
Render image submit form in modal

### LINKED ISSUE #5: Move Save Image to a Modal
#### As a user, I want my save image submission form in a modal.

__Story__:

Now that we are displaying images for users and allowing them to submit on the same page, our website has become a bit
crowded. In order to fix this problem, User Experience (UX) has suggested we move the submission form into a modal to
clean up the page. In order to solve this, we will use `react-gears`.

__Acceptance criteria__:
- [x] Ran the command `yarn install reactstrap` in their project.
- [x] The landing page shows a link to open the submission form.
- [x] Use Reactstrap for your modal.
- [x] Submission form is now in a modal.
- [x] I cannot successfully save an image with an invalid URL on the front-end.
- [x] I cannot successfully save an image with an invalid URL on the backend.
- [x] An error message is associated with the appropriate input field on
  failure.
- [x] The successfully saved image is displayed on the landing page.

__Dependencies__:
- Save Image Link Part 2

__Reference material__: 
- https://reactstrap.github.io/components/
- https://wesbos.com/javascript-modules/

### PR DESCRIPTION
**Pull request by tyleralves**
Closes #5 

https://image-sharer-tyleralvess-pr-19.herokuapp.com/

### CODE REVIEW

**Comment by tyleralves** [comment]
[Ropes #5](https://trello.com/c/9pBrooAP/613-ropes-5)

File: app/javascript/scenes/Landing.jsx [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
   state = {
     linkInput: '',
     images: [],
+    isModalOpen: false
   };
 
   constructor(props) {
     super(props);
     this.onSubmit = this.onSubmit.bind(this);
+    this.onModalToggle = this.onModalToggle.bind(this);
```
Comment (ramonrovirosa) [inline app/javascript/scenes/Landing.jsx]: If we used arrow functions for these methods we could get rid of the constructor all together.

File: app/assets/stylesheets/application.css [hunk 2]
```diff
@@ -14,5 +14,9 @@
  *= require_self
  */
 .landing-img {
-  width: 400px;
+  max-width: 400px;
```
Comment (ramonrovirosa) [inline app/assets/stylesheets/application.css]: I'm glad you did this!
I was going to mention in your previous PR that thinking responsive/mobile is good!

File: app/javascript/scenes/Landing.jsx (hunk 1 above)
Comment (tyleralves) [reply app/javascript/scenes/Landing.jsx]: I absolutely never do this, but I was trying to spy on them with Enzyme and the Airbnb team said they wouldn't even attempt to support spying on arrow function class methods.

Let me know what you think:
https://github.com/airbnb/enzyme/issues/365

Anyway, I took out the spies so I switched them back to arrows.

Comment (ramonrovirosa) [reply app/javascript/scenes/Landing.jsx]: Oh interesting....thanks for sharing, I wasn't aware that was an issue. Thats unfortunate!

**Pull request merged**